src/abe_policy/dimension.rs

```diff
@@ -265,17 +268,23 @@ impl Dimension {
         new_name: &str,
     ) -> Result<(), Error> {
         if self.attributes.contains_key(new_name) {
-            Err(Error::OperationNotPermitted(
+            return Err(Error::OperationNotPermitted(
                 "New attribute name is already used in the same dimension".to_string(),
-            ))
-        } else {
-            match self.attributes.remove(attr_name) {
-                Some(attr_params) => {
-                    self.attributes.insert(new_name.to_string(), attr_params);
-                    Ok(())
+            ));
+        }
+        match self.attributes.remove(attr_name) {
+            Some(attr_params) => {
+                self.attributes.insert(new_name.to_string(), attr_params);
+                if let Some(order) = self.order.as_mut() {
+                    order.iter_mut().for_each(|name| {
+                        if name == attr_name {
+                            *name = new_name.to_string()
```

Comment:
You may not need to store owned string object, but a reference to attribute names stored in the `self.attributes`. This would be a way to avoid cloning here (plus take ownership of `new_name` in this function as you need it).